core/src/services/supabase/backend.rs

```diff
@@ -0,0 +1,342 @@
+// Licensed to the Apache Software Foundation (ASF) under one
+// or more contributor license agreements.  See the NOTICE file
+// distributed with this work for additional information
+// regarding copyright ownership.  The ASF licenses this file
+// to you under the Apache License, Version 2.0 (the
+// "License"); you may not use this file except in compliance
+// with the License.  You may obtain a copy of the License at
+//
+//   http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing,
+// software distributed under the License is distributed on an
+// "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, either express or implied.  See the License for the
+// specific language governing permissions and limitations
+// under the License.
+
+use std::fmt::Debug;
+use std::sync::Arc;
+
+use async_trait::async_trait;
+use http::HeaderValue;
+use http::StatusCode;
+use log::debug;
+
+use super::core::*;
+use super::error::parse_error;
+use super::writer::*;
+use crate::ops::*;
+use crate::raw::*;
+use crate::*;
+
+/// Supabase service
+///
+/// # Capabilities
+///
+/// - [x] read
+/// - [x] write
+/// - [ ] copy
+/// - [ ] list
+/// - [ ] scan
+/// - [ ] presign
+/// - [ ] blocking
+///
+/// # Configuration
+///
+/// - `root`: Set the work dir for backend.
+/// - `bucket`: Set the container name for backend.
+/// - `endpoint`: Set the endpoint for backend.
+/// - `service_key`: Set the secret key for the backend, this is prioritized over the client key
+/// - `anon_key`: Set the client key for the backend, this cannot access non-public resources
+///
+/// ## Authorization key setup
+///
+/// There are two types of key in the Supabase, one is anon_key(Client key), another one is
+/// service_role_key(Secret key). The former one can only access public resources while the latter one
+/// can access all resources.
+///
+/// There exists a priority of setting up the authorization keys.
+/// The key set up by builder takes precedence over environment variable, and the service_key
+/// takes precedence over anon_key.
+///
+/// There are two ways of setting up the authorization key, i.e., environment variable and builder.
+/// - The environment variable OPENDAL_SUPABASE_AUTH_ANON_KEY sets up the anon_key.
```

Comment:
We should only mention env here is service has its own env, for example: AWS_ACCESS_KEY_ID. And `OPENDAL_` env are reserved as internal usage which could be changed at anytime.

So we should not mention the env way here.